fixed: Media Library - Can`t delete empty folder #17263

noobCode-69 opened the pull request:
### What does it do?
The commit fixes the issue of not being able to delete folders in the media library 

**Earlier Behaviour**
- Go to media library
- Create a folder
- Click on "pen" icon to edit then click "Delete folder"
- A modal will appear click "Confirm" on it
- You will see nothing happens

**Wanted behaviour**
- Folder gets deleted when clicking "Confirm"
### Why is it needed?
Because we want to delete folders using the above steps.

noobCode-69 commented:
Hey @gu-stav  please review and merge.

joshuaellis commented:
Can you fill in the description accurately please, "What does it do" is far too light imo.

Also please try to avoid pinging members :)

noobCode-69 commented:
How I fixed it
**Problem**
In the "EditFolderDialog.js" you will see logic in the return state
```js
<>
    <ModalLayout>
        // some code
   </ModlaLayout>
    {showConfirmDialog && (
        <RemoveFolderDialog onClose={() => setShowConfirmDialog(false)} onConfirm={handleDelete} />
     )}
</>
```
**Fix**
The problem is related to event propagation , since you will see if you click **anywhere** in the modal the modal disappears
- This is how i fixed it
```js
<ModalLayout>
   // some code
   {showConfirmDialog && (
      <RemoveFolderDialog onClose={() => setShowConfirmDialog(false)} onConfirm={handleDelete} />
   )}
</ModlaLayout>   
```

noobCode-69 commented:
Hey @Eventyret can you please review this, I think this solves the problem

jhoward1994 reviewed (approved):
Thanks for this fix!
